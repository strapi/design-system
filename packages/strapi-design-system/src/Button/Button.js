import React from 'react';

import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import Loader from '@strapi/icons/Loader';
import { Typography } from '../Typography';
import { Box } from '../Box';
import { getDisabledStyle, getHoverStyle, getActiveStyle, getVariantStyle } from './utils';
import { VARIANTS, BUTTON_SIZES } from './constants';
import { BaseButton } from '../BaseButton';

const rotation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`;

const LoadingWrapper = styled.div`
  animation: ${rotation} 2s infinite linear;
  will-change: transform;
`;

const BoxFullHeight = styled(Box)`
  height: 100%;
`;

export const ButtonWrapper = styled(BaseButton)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.buttonPrimary600};
  border: 1px solid ${({ theme }) => theme.colors.buttonPrimary600};
  height: ${({ theme, size }) => theme.sizes.button[size]};
  padding-left: ${({ theme }) => theme.spaces[4]};
  padding-right: ${({ theme }) => theme.spaces[4]};

  ${Box} {
    display: flex;
    align-items: center;
  }

  ${Typography} {
    color: ${({ theme }) => theme.colors.buttonNeutral0};
  }

  &[aria-disabled='true'] {
    ${getDisabledStyle}
    &:active {
      ${getDisabledStyle}
    }
  }
  &:hover {
    ${getHoverStyle}
  }
  &:active {
    ${getActiveStyle}
  }
  ${getVariantStyle}
  ${({ fullWidth }) =>
    fullWidth &&
    `
    display: inline-flex;
    justify-content: center;
    width: 100%;
  `}
`;

export const Button = React.forwardRef(
  ({ variant, startIcon, endIcon, disabled, children, onClick, size, loading, fullWidth, ...props }, ref) => {
    const isDisabled = disabled || loading;

    const handleClick = (e) => {
      if (!isDisabled && onClick) {
        onClick(e);
      }
    };

    return (
      <ButtonWrapper
        ref={ref}
        aria-disabled={isDisabled}
        disabled={isDisabled}
        size={size}
        variant={variant}
        onClick={handleClick}
        fullWidth={fullWidth}
        {...props}
      >
        {(startIcon || loading) && (
          <BoxFullHeight aria-hidden paddingRight={2}>
            {loading ? (
              <LoadingWrapper>
                <Loader />
              </LoadingWrapper>
            ) : (
              startIcon
            )}
          </BoxFullHeight>
        )}

        <Typography variant={size === 'S' ? 'pi' : undefined} fontWeight="bold" lineHeight={0}>
          {children}
        </Typography>

        {endIcon && (
          <Box aria-hidden paddingLeft={2}>
            {endIcon}
          </Box>
        )}
      </ButtonWrapper>
    );
  },
);

Button.displayName = 'Button';

Button.defaultProps = {
  disabled: false,
  endIcon: undefined,
  fullWidth: false,
  loading: false,
  onClick: undefined,
  size: 'S',
  startIcon: undefined,
  variant: 'default',
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  endIcon: PropTypes.element,
  fullWidth: PropTypes.bool,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
  size: PropTypes.oneOf(BUTTON_SIZES),
  startIcon: PropTypes.element,
  variant: PropTypes.oneOf(VARIANTS),
};
