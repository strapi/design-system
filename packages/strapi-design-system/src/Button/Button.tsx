import React from 'react';

import { Loader } from '@strapi/icons';
import styled, { keyframes } from 'styled-components';

import { BUTTON_SIZES, Variant, ButtonSizes, DEFAULT } from './constants';
import { getDisabledStyle, getHoverStyle, getActiveStyle, getVariantStyle } from './utils';
import { BaseButton, BaseButtonProps } from '../BaseButton';
import { Box } from '../Box';
import { Flex } from '../Flex';
import { Typography } from '../Typography';

const rotation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`;

const LoaderAnimated = styled(Loader)`
  animation: ${rotation} 2s infinite linear;
  will-change: transform;
`;

export const ButtonWrapper = styled(BaseButton)<Required<Pick<ButtonProps, 'size' | 'variant'>>>`
  height: ${({ theme, size }) => theme.sizes.button[size]};

  svg {
    height: ${12 / 16}rem;
    width: auto;
  }

  &[aria-disabled='true'] {
    ${getDisabledStyle}

    &:active {
      ${getDisabledStyle}
    }
  }

  &:hover {
    ${getHoverStyle}
  }

  &:active {
    ${getActiveStyle}
  }

  ${getVariantStyle}
`;

export interface ButtonProps extends BaseButtonProps {
  endIcon?: React.ReactNode;
  fullWidth?: boolean;
  loading?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  size?: ButtonSizes;
  startIcon?: React.ReactNode;
  variant?: Variant;
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      variant = DEFAULT,
      startIcon,
      endIcon,
      disabled = false,
      children,
      onClick,
      size = BUTTON_SIZES[0],
      loading = false,
      fullWidth = false,
      ...props
    },
    ref,
  ) => {
    const isDisabled = disabled || loading;

    const handleClick: React.MouseEventHandler<HTMLButtonElement> = (e) => {
      if (!isDisabled && onClick) {
        onClick(e);
      }
    };

    return (
      <ButtonWrapper
        ref={ref}
        aria-disabled={isDisabled}
        disabled={isDisabled}
        size={size}
        variant={variant}
        onClick={handleClick}
        fullWidth={fullWidth}
        alignItems="center"
        background="buttonPrimary600"
        borderColor="buttonPrimary600"
        gap={2}
        inline={fullWidth}
        justifyContent={fullWidth ? 'center' : undefined}
        paddingLeft={4}
        paddingRight={4}
        width={fullWidth ? '100%' : undefined}
        {...props}
      >
        {(startIcon || loading) && <Box aria-hidden>{loading ? <LoaderAnimated /> : startIcon}</Box>}

        <Typography
          variant={size === 'S' ? 'pi' : undefined}
          fontWeight="bold"
          lineHeight={0}
          textColor="buttonNeutral0"
        >
          {children}
        </Typography>

        {endIcon && <Flex aria-hidden>{endIcon}</Flex>}
      </ButtonWrapper>
    );
  },
);

Button.displayName = 'Button';
