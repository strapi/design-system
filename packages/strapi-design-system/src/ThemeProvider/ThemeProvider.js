import * as React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider, createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
 /* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
  }
  /* HTML5 display-role reset for older browsers */
  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }
  body {
    line-height: 1;
  }
  ol,
  ul {
    list-style: none;
  }
  blockquote,
  q {
    quotes: none;
  }
  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }
  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
  /* My styles */
  *,
  *:before,
  *:after {
    box-sizing: border-box;
  }

  *:focus-visible {
    box-shadow: 0px 0px 6px ${(props) => props.theme.colors.primary500};
    // Windows High Contrast won't see the bow shadow but transparent will
    // be replaced by a color
    // See https://sarahmhigley.com/writing/whcm-quick-tips/ for more details
    outline: 4px solid transparent;
  }
  /* Focusing the button with a mouse, touch, or stylus will show a subtle drop shadow. */
  *:focus:not(:focus-visible) {
    outline: none;
  }

  body,html{
    height: 100%;
  }
  body {
    font-family: --apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  }
`;

export const ThemeProvider = ({ children, theme }) => {
  return (
    <StyledThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
  theme: PropTypes.object.isRequired,
};
