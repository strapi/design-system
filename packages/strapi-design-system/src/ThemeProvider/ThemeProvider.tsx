/**
 * TODO: remove this in v2 in favour of DesignSystemProvider, but this is rather convienient for now.
 */
import * as React from 'react';

import { ThemeProvider as StyledThemeProvider, createGlobalStyle, DefaultTheme } from 'styled-components';

import { LiveRegions } from '../LiveRegions/LiveRegions';
import { lightTheme } from '../themes';

const GlobalStyle = createGlobalStyle`
 /* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
  }
  /* HTML5 display-role reset for older browsers */
  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }
  body {
    line-height: 1;
  }
  ol,
  ul {
    list-style: none;
  }
  blockquote,
  q {
    quotes: none;
  }
  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }
  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
  /* My styles */
  *,
  *:before,
  *:after {
    box-sizing: border-box;
  }

  *:focus-visible {
    outline: 2px solid ${({ theme }) => theme.colors.primary600};
    outline-offset: 2px;
  }

  /* Focusing the button with a mouse, touch, or stylus will show a subtle drop shadow. */
  *:focus:not(:focus-visible) {
    outline: none;
  }

  body,html{
    height: 100%;
  }
  body {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  }

  button {
    margin: 0;
    padding: 0;
    border: none;
    background: none;
    color: inherit;
    font-size: 100%;
    cursor: pointer;
    font: inherit;
  }

  textarea,input {
    font: inherit;
  }

  .lock-body-scroll {
    height: 100vh;
    overflow-y: hidden;
  }
`;

export interface ThemeProviderProps {
  children: React.ReactNode;
  theme?: DefaultTheme;
}

export const ThemeProvider = ({ children, theme = lightTheme }: ThemeProviderProps) => {
  return (
    <StyledThemeProvider theme={theme}>
      {children}
      <LiveRegions />
      <GlobalStyle />
    </StyledThemeProvider>
  );
};
