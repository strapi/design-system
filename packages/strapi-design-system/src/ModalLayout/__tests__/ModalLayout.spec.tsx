import { render } from '@testing-library/react';

import { Button } from '../../Button';
import { ThemeProvider } from '../../ThemeProvider';
import { lightTheme } from '../../themes';
import { ModalLayout, ModalHeader, ModalBody, ModalFooter } from '../index';

describe('ModalLayout', () => {
  it('should render component and match snapshot', () => {
    render(
      <ThemeProvider theme={lightTheme}>
        <ModalLayout onClose={() => jest.fn()} labelledBy="title">
          <ModalHeader>Modal Title</ModalHeader>
          <ModalBody>Hello World</ModalBody>
          <ModalFooter
            startActions={
              <Button onClick={() => jest.fn} variant="tertiary">
                Cancel
              </Button>
            }
            endActions={
              <>
                <Button variant="secondary">Add new stuff</Button>
                <Button onClick={() => jest.fn()}>Finish</Button>
              </>
            }
          />
        </ModalLayout>
      </ThemeProvider>,
    );

    expect(document.body).toMatchInlineSnapshot(`
      .c21 {
        font-size: 0.75rem;
        line-height: 1.33;
        font-weight: 600;
        line-height: 0;
        color: #ffffff;
      }

      .c1 {
        padding-right: 40px;
        padding-left: 40px;
        position: fixed;
        z-index: 4;
      }

      .c4 {
        background: #ffffff;
        border-radius: 4px;
        box-shadow: 0px 2px 15px rgba(33,33,52,0.1);
      }

      .c6 {
        background: #f6f6f9;
        padding-top: 16px;
        padding-right: 20px;
        padding-bottom: 16px;
        padding-left: 20px;
      }

      .c9 {
        background: #ffffff;
        padding: 8px;
        border-radius: 4px;
        border-color: #dcdce4;
        border: 1px solid #dcdce4;
        width: 2rem;
        height: 2rem;
        cursor: pointer;
      }

      .c12 {
        padding: 32px;
      }

      .c17 {
        background: #4945ff;
        padding: 8px;
        padding-right: 16px;
        padding-left: 16px;
        border-radius: 4px;
        border-color: #4945ff;
        border: 1px solid #4945ff;
        cursor: pointer;
      }

      .c2 {
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-box-pack: center;
        -webkit-justify-content: center;
        -ms-flex-pack: center;
        justify-content: center;
      }

      .c8 {
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-box-pack: justify;
        -webkit-justify-content: space-between;
        -ms-flex-pack: justify;
        justify-content: space-between;
      }

      .c15 {
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
      }

      .c18 {
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        gap: 8px;
      }

      .c10 {
        position: relative;
        outline: none;
      }

      .c10 > svg {
        height: 12px;
        width: 12px;
      }

      .c10 > svg > g,
      .c10 > svg path {
        fill: #ffffff;
      }

      .c10[aria-disabled='true'] {
        pointer-events: none;
      }

      .c10:after {
        -webkit-transition-property: all;
        transition-property: all;
        -webkit-transition-duration: 0.2s;
        transition-duration: 0.2s;
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -4px;
        bottom: -4px;
        left: -4px;
        right: -4px;
        border: 2px solid transparent;
      }

      .c10:focus-visible {
        outline: none;
      }

      .c10:focus-visible:after {
        border-radius: 8px;
        content: '';
        position: absolute;
        top: -5px;
        bottom: -5px;
        left: -5px;
        right: -5px;
        border: 2px solid #4945ff;
      }

      .c19 {
        height: 2rem;
        border: 1px solid #dcdce4;
        background: #ffffff;
      }

      .c19[aria-disabled='true'] {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c19[aria-disabled='true'] .c20 {
        color: #666687;
      }

      .c19[aria-disabled='true'] svg > g,
      .c19[aria-disabled='true'] svg path {
        fill: #666687;
      }

      .c19[aria-disabled='true']:active {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c19[aria-disabled='true']:active .c20 {
        color: #666687;
      }

      .c19[aria-disabled='true']:active svg > g,
      .c19[aria-disabled='true']:active svg path {
        fill: #666687;
      }

      .c19:hover {
        background-color: #f6f6f9;
      }

      .c19:active {
        background-color: #eaeaef;
      }

      .c19 .c20 {
        color: #32324d;
      }

      .c19 svg > g,
      .c19 svg path {
        fill: #32324d;
      }

      .c22 {
        height: 2rem;
        border: 1px solid #d9d8ff;
        background: #f0f0ff;
      }

      .c22[aria-disabled='true'] {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c22[aria-disabled='true'] .c20 {
        color: #666687;
      }

      .c22[aria-disabled='true'] svg > g,
      .c22[aria-disabled='true'] svg path {
        fill: #666687;
      }

      .c22[aria-disabled='true']:active {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c22[aria-disabled='true']:active .c20 {
        color: #666687;
      }

      .c22[aria-disabled='true']:active svg > g,
      .c22[aria-disabled='true']:active svg path {
        fill: #666687;
      }

      .c22:hover {
        background-color: #ffffff;
      }

      .c22:active {
        background-color: #ffffff;
        border: 1px solid #4945ff;
      }

      .c22:active .c20 {
        color: #4945ff;
      }

      .c22:active svg > g,
      .c22:active svg path {
        fill: #4945ff;
      }

      .c22 .c20 {
        color: #271fe0;
      }

      .c22 svg > g,
      .c22 svg path {
        fill: #271fe0;
      }

      .c23 {
        height: 2rem;
      }

      .c23[aria-disabled='true'] {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c23[aria-disabled='true'] .c20 {
        color: #666687;
      }

      .c23[aria-disabled='true'] svg > g,
      .c23[aria-disabled='true'] svg path {
        fill: #666687;
      }

      .c23[aria-disabled='true']:active {
        border: 1px solid #dcdce4;
        background: #eaeaef;
      }

      .c23[aria-disabled='true']:active .c20 {
        color: #666687;
      }

      .c23[aria-disabled='true']:active svg > g,
      .c23[aria-disabled='true']:active svg path {
        fill: #666687;
      }

      .c23:hover {
        border: 1px solid #7b79ff;
        background: #7b79ff;
      }

      .c23:active {
        border: 1px solid #4945ff;
        background: #4945ff;
      }

      .c23 svg > g,
      .c23 svg path {
        fill: #ffffff;
      }

      .c0 {
        border: 0;
        -webkit-clip: rect(0 0 0 0);
        clip: rect(0 0 0 0);
        height: 1px;
        margin: -1px;
        overflow: hidden;
        padding: 0;
        position: absolute;
        width: 1px;
      }

      .c3 {
        inset: 0;
        background: #32324d1F;
      }

      .c5 {
        width: 51.875rem;
      }

      .c11 svg > g,
      .c11 svg path {
        fill: #8e8ea9;
      }

      .c11:hover svg > g,
      .c11:hover svg path {
        fill: #666687;
      }

      .c11:active svg > g,
      .c11:active svg path {
        fill: #a5a5ba;
      }

      .c11[aria-disabled='true'] svg path {
        fill: #666687;
      }

      .c7 {
        border-radius: 4px 4px 0 0;
        border-bottom: 1px solid #eaeaef;
      }

      .c14 {
        border-radius: 0 0 4px 4px;
        border-top: 1px solid #eaeaef;
      }

      .c16 > * + * {
        margin-left: 8px;
      }

      .c13 {
        overflow: auto;
        max-height: 60vh;
      }

      <body
        class="lock-body-scroll"
      >
        <div>
          <div
            class="c0"
          >
            <p
              aria-live="polite"
              aria-relevant="all"
              id="live-region-log"
              role="log"
            />
            <p
              aria-live="polite"
              aria-relevant="all"
              id="live-region-status"
              role="status"
            />
            <p
              aria-live="assertive"
              aria-relevant="all"
              id="live-region-alert"
              role="alert"
            />
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="c1 c2 c3"
          >
            <div>
              <div>
                <div
                  aria-labelledby="title"
                  aria-modal="true"
                  class="c4 c5"
                  role="dialog"
                >
                  <div
                    class="c6 c7"
                  >
                    <div
                      class="c8"
                    >
                      Modal Title
                      <button
                        aria-disabled="false"
                        class="c9 c2 c10 c11"
                        type="button"
                      >
                        <span
                          class="c0"
                        >
                          Close the modal
                        </span>
                        <svg
                          aria-hidden="true"
                          fill="none"
                          focusable="false"
                          height="1rem"
                          viewBox="0 0 24 24"
                          width="1rem"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M24 2.417 21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417Z"
                            fill="#212134"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="c12 c13"
                  >
                    Hello World
                  </div>
                  <div
                    class="c6 c14"
                  >
                    <div
                      class="c8"
                    >
                      <div
                        class="c15 c16"
                      >
                        <button
                          aria-disabled="false"
                          class="c17 c18 c10 c19"
                          type="button"
                        >
                          <span
                            class="c20 c21"
                          >
                            Cancel
                          </span>
                        </button>
                      </div>
                      <div
                        class="c15 c16"
                      >
                        <button
                          aria-disabled="false"
                          class="c17 c18 c10 c22"
                          type="button"
                        >
                          <span
                            class="c20 c21"
                          >
                            Add new stuff
                          </span>
                        </button>
                        <button
                          aria-disabled="false"
                          class="c17 c18 c10 c23"
                          type="button"
                        >
                          <span
                            class="c20 c21"
                          >
                            Finish
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    `);
  });
});
