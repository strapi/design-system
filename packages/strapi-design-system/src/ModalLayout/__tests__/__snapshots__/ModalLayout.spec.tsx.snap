// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLayout should render component and match snapshot 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c18 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c1 {
  padding-right: 40px;
  padding-left: 40px;
  position: fixed;
  z-index: 4;
}

.c4 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 2px 15px rgba(33,33,52,0.1);
  width: 51.875rem;
}

.c5 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c8 {
  background: #ffffff;
  padding: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
  cursor: pointer;
}

.c11 {
  padding: 32px;
}

.c15 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c9 {
  position: relative;
  outline: none;
}

.c9 > svg {
  height: 12px;
  width: 12px;
}

.c9 > svg > g,
.c9 > svg path {
  fill: #ffffff;
}

.c9[aria-disabled='true'] {
  pointer-events: none;
}

.c9:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c16 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c16 svg {
  height: 0.75rem;
  width: auto;
}

.c16[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true'] .c17 {
  color: #666687;
}

.c16[aria-disabled='true'] svg > g,
.c16[aria-disabled='true'] svg path {
  fill: #666687;
}

.c16[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c16[aria-disabled='true']:active svg > g,
.c16[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c16:hover {
  background-color: #f6f6f9;
}

.c16:active {
  background-color: #eaeaef;
}

.c16 .c17 {
  color: #32324d;
}

.c16 svg > g,
.c16 svg path {
  fill: #32324d;
}

.c19 {
  height: 2rem;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c19 svg {
  height: 0.75rem;
  width: auto;
}

.c19[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c19[aria-disabled='true'] .c17 {
  color: #666687;
}

.c19[aria-disabled='true'] svg > g,
.c19[aria-disabled='true'] svg path {
  fill: #666687;
}

.c19[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c19[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c19[aria-disabled='true']:active svg > g,
.c19[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c19:hover {
  background-color: #ffffff;
}

.c19:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c19:active .c17 {
  color: #4945ff;
}

.c19:active svg > g,
.c19:active svg path {
  fill: #4945ff;
}

.c19 .c17 {
  color: #271fe0;
}

.c19 svg > g,
.c19 svg path {
  fill: #271fe0;
}

.c20 {
  height: 2rem;
}

.c20 svg {
  height: 0.75rem;
  width: auto;
}

.c20[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c20[aria-disabled='true'] .c17 {
  color: #666687;
}

.c20[aria-disabled='true'] svg > g,
.c20[aria-disabled='true'] svg path {
  fill: #666687;
}

.c20[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c20[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c20[aria-disabled='true']:active svg > g,
.c20[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c20:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c20:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c20 svg > g,
.c20 svg path {
  fill: #ffffff;
}

.c3 {
  background: #32324d1F;
  inset: 0;
}

.c10 {
  border-color: #dcdce4;
  height: 2rem;
  width: 2rem;
}

.c10 svg g,
.c10 svg path {
  fill: #8e8ea9;
}

.c10:hover svg g,
.c10:focus svg g,
.c10:hover svg path,
.c10:focus svg path {
  fill: #666687;
}

.c10[aria-disabled='true'] svg path {
  fill: #666687;
}

.c6 {
  border-radius: 4px 4px 0 0;
  border-bottom: 1px solid #eaeaef;
}

.c13 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c12 {
  overflow: auto;
  max-height: 60vh;
}

<body
  class="lock-body-scroll"
>
  <div>
    <div
      class="c0"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
  <div
    class=""
  >
    <div
      class="c1 c2 c3"
    >
      <div>
        <div>
          <div
            aria-labelledby="title"
            aria-modal="true"
            class="c4"
            role="dialog"
          >
            <div
              class="c5 c6"
            >
              <div
                class="c7"
              >
                Modal Title
                <button
                  aria-disabled="false"
                  class="c8 c2 c9 c10"
                  type="button"
                >
                  <span
                    class="c0"
                  >
                    Close the modal
                  </span>
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="1rem"
                    viewBox="0 0 24 24"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 2.417 21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417Z"
                      fill="#212134"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="c11 c12"
            >
              Hello World
            </div>
            <div
              class="c5 c13"
            >
              <div
                class="c7"
              >
                <div
                  class="c14"
                >
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c16"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Cancel
                    </span>
                  </button>
                </div>
                <div
                  class="c14"
                >
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c19"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Add new stuff
                    </span>
                  </button>
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c20"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Finish
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
